src/controllers/product.controller.ts

```diff
@@ -62,4 +62,29 @@ const getFiltered = async (
   }
 };
 
-export default { getAll, getOne, getFiltered };
+const getRecommended = async (req: Request, res: Response) => {
+  const productId = req.params.id;
+
+  try {
+    const foundProduct = await productsService.getOne(productId);
+
+    if (foundProduct) {
+      const allProducts = await productsService.allProducts();
+
+      const randomProducts = productsService.getRandom(allProducts, 10);
```

Comment:
I would move getting all products to service, in controller you only receive the random products
```suggestion
      const randomProducts = productsService.getRandom(10);
```